string[] effector
uint16[] target
bool[] isTouching